### PR TITLE
Destinations v2 specs: add raw namespace override; add ordering

### PR DESCRIPTION
**Pull request by edgao**
based on https://github.com/airbytehq/airbyte/pull/27845.

* Add a raw_data_dataset / raw_data_schema option
* add explicit `order` tags + move the v2 options to the bottom of the spec

do not merge until after https://github.com/airbytehq/airbyte/pull/28723 is merged

### CODE REVIEW

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`8ce5b04b85`](https://github.com/airbytehq/airbyte/pull/28744/commits/8ce5b04b85a78e95d76604c54a81aa2570d40505)) - ❌

⏲️  Total pipeline duration: 38mn07s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690393763/8ce5b04b85a78e95d76604c54a81aa2570d40505/destination-bigquery/1.5.7/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`8ce5b04b85`](https://github.com/airbytehq/airbyte/pull/28744/commits/8ce5b04b85a78e95d76604c54a81aa2570d40505)) - ✅

⏲️  Total pipeline duration: 21mn46s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690393763/8ce5b04b85a78e95d76604c54a81aa2570d40505/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`8ce5b04b85`](https://github.com/airbytehq/airbyte/pull/28744/commits/8ce5b04b85a78e95d76604c54a81aa2570d40505)) - ❌

⏲️  Total pipeline duration: 60mn03s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690393763/8ce5b04b85a78e95d76604c54a81aa2570d40505/destination-snowflake/1.2.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json [hunk 1]
```diff
@@ -211,6 +206,18 @@
...
+      "use_1s1t_format": {
+        "type": "boolean",
+        "description": "(Beta) Use <a href=\"https://github.com/airbytehq/airbyte/issues/26028\" target=\"_blank\">Destinations V2</a>. Contact Airbyte Support to participate in the beta program.",
+        "title": "Use Destinations V2 (Early Access)",
+        "order": 7
+      },
+      "raw_data_dataset": {
+        "type": "string",
+        "description": "(Beta) The dataset to write raw tables into",
+        "title": "Destinations V2 raw table dataset (Early Access)",
```
Comment (cynthiaxyin) [inline airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json:219]: nit: Should these be title case?

**Review by cynthiaxyin — APPROVED** [review]
Can you remind me if/where we discussed deciding to reimplement this?

**Comment by edgao** [comment]
https://github.com/airbytehq/airbyte/issues/28403 includes reenabling the spec option if it's low-effort

> Add (back?) a spec option of the destination which allows this internal namespace to be overwritten.

> If allowing overrides of the namespace adds significant effort, we should not do it now.

File: airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json (hunk 1 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json]: yep, fixed.

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`b364c8e318`](https://github.com/airbytehq/airbyte/pull/28744/commits/b364c8e318ca52f2eafd94af7e2c1daf4a10901f)) - ✅

⏲️  Total pipeline duration: 22mn17s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690470626/b364c8e318ca52f2eafd94af7e2c1daf4a10901f/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`b364c8e318`](https://github.com/airbytehq/airbyte/pull/28744/commits/b364c8e318ca52f2eafd94af7e2c1daf4a10901f)) - ❌

⏲️  Total pipeline duration: 37mn56s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ❌        |
| Connector version increment check                                               | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690470626/b364c8e318ca52f2eafd94af7e2c1daf4a10901f/destination-bigquery/1.5.7/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Review by jbfbell — APPROVED** [review]

File: airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json [hunk 2]
```diff
@@ -211,6 +206,18 @@
...
+      },
+      "use_1s1t_format": {
+        "type": "boolean",
+        "description": "(Beta) Use <a href=\"https://github.com/airbytehq/airbyte/issues/26028\" target=\"_blank\">Destinations V2</a>. Contact Airbyte Support to participate in the beta program.",
+        "title": "Use Destinations V2 (Early Access)",
+        "order": 7
+      },
+      "raw_data_dataset": {
+        "type": "string",
+        "description": "(Beta) The dataset to write raw tables into",
```
Comment (cynthiaxyin) [inline airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json:218]: Doesn't have to be in this change, but should we eventually have a longer description about why a user might want to put something here?

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`b364c8e318`](https://github.com/airbytehq/airbyte/pull/28744/commits/b364c8e318ca52f2eafd94af7e2c1daf4a10901f)) - ❌

⏲️  Total pipeline duration: 63mn03s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690470626/b364c8e318ca52f2eafd94af7e2c1daf4a10901f/destination-snowflake/1.2.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/tools/ci_connector_ops/ci_connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

File: airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json (hunk 2 above)
Comment (edgao) [reply airbyte-integrations/connectors/destination-bigquery/src/main/resources/spec.json:218]: yeah, I figure we can workshop the wording closer to full release. there's probably a lot of stuff that we should polish here, e.g. adding default values and stuff :shrug:

**Comment by edgao** [comment]
switching back to draft mode to avoid further connectors CI stuff

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`46a8c6a07e`](https://github.com/airbytehq/airbyte/pull/28744/commits/46a8c6a07e2c01e123714c9a1c8ea910375729f2)) - ✅

⏲️  Total pipeline duration: 20mn06s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690480018/46a8c6a07e2c01e123714c9a1c8ea910375729f2/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`46a8c6a07e`](https://github.com/airbytehq/airbyte/pull/28744/commits/46a8c6a07e2c01e123714c9a1c8ea910375729f2)) - ❌

⏲️  Total pipeline duration: 31mn20s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690480018/46a8c6a07e2c01e123714c9a1c8ea910375729f2/destination-bigquery/1.6.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-snowflake test report (commit [`46a8c6a07e`](https://github.com/airbytehq/airbyte/pull/28744/commits/46a8c6a07e2c01e123714c9a1c8ea910375729f2)) - ❌

⏲️  Total pipeline duration: 44mn19s 

| Step                                                                             | Result   |
|:---------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-snowflake/metadata.yaml     | ✅        |
| Connector version semver check                                                   | ✅        |
| Connector version increment check                                                | ❌        |
| QA checks                                                                        | ✅        |
| Build connector tar                                                              | ✅        |
| Build destination-snowflake docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization-snowflake:dev                                        | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-snowflake:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690480018/46a8c6a07e2c01e123714c9a1c8ea910375729f2/destination-snowflake/1.2.4/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-snowflake test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery test report (commit [`8c056a862c`](https://github.com/airbytehq/airbyte/pull/28744/commits/8c056a862c524341380dcdf721196dcbc2780c8c)) - ❌

⏲️  Total pipeline duration: 36mn02s 

| Step                                                                            | Result   |
|:--------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery/metadata.yaml     | ✅        |
| Connector version semver check                                                  | ✅        |
| Connector version increment check                                               | ❌        |
| QA checks                                                                       | ✅        |
| Build connector tar                                                             | ✅        |
| Build destination-bigquery docker image for platform linux/x86_64               | ✅        |
| Build airbyte/normalization:dev                                                 | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery:integrationTest | ❌        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690485917/8c056a862c524341380dcdf721196dcbc2780c8c/destination-bigquery/1.6.0/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery test
```

**Comment by octavia-squidington-iii** [comment]
##  destination-bigquery-denormalized test report (commit [`8c056a862c`](https://github.com/airbytehq/airbyte/pull/28744/commits/8c056a862c524341380dcdf721196dcbc2780c8c)) - ✅

⏲️  Total pipeline duration: 02mn39s 

| Step                                                                                         | Result   |
|:---------------------------------------------------------------------------------------------|:---------|
| Validate airbyte-integrations/connectors/destination-bigquery-denormalized/metadata.yaml     | ✅        |
| Connector version semver check                                                               | ✅        |
| QA checks                                                                                    | ✅        |
| Build connector tar                                                                          | ✅        |
| Build destination-bigquery-denormalized docker image for platform linux/x86_64               | ✅        |
| ./gradlew :airbyte-integrations:connectors:destination-bigquery-denormalized:integrationTest | ✅        |

🔗 [View the logs here](https://storage.cloud.google.com/airbyte-ci-reports-multi/airbyte-ci/connectors/test/pull_request/edgao_destinationsv2_beta_spec_updates/1690485917/8c056a862c524341380dcdf721196dcbc2780c8c/destination-bigquery-denormalized/1.5.3/output.html)

*Please note that tests are only run on PR ready for review. Please set your PR to draft mode to not flood the CI engine and upstream service on following commits.*
**You can run the same pipeline locally on this branch with the [airbyte-ci](https://github.com/airbytehq/airbyte/blob/master/airbyte-ci/connector_ops/connector_ops/pipelines/README.md) tool with the following command**
```bash
airbyte-ci connectors --name=destination-bigquery-denormalized test
```

**Pull request merged**